lib/gui/app/components/drive-selector/DriveSelectorModal.jsx

```diff
@@ -0,0 +1,320 @@
+   *        if (shouldChangeDriveSelectionState) doSomething();
+   *    });
+   */
+  const shouldChangeDriveSelectionState = (drive) => {
+    return isDriveValid(drive, selectionState.getImage())
+  }
+
+  /**
+   * @summary Toggle a drive selection
+   * @function
+   * @public
+   *
+   * @param {Object} drive - drive
+   * @returns {void}
+   *
+   * @example
+   * toggleDrive({
+   *   device: '/dev/disk2',
+   *   size: 999999999,
+   *   name: 'Cruzer USB drive'
+   * });
+   */
+  const toggleDrive = (drive) => {
+    const canChangeDriveSelectionState = shouldChangeDriveSelectionState(drive)
+
+    if (canChangeDriveSelectionState) {
+      analytics.logEvent('Toggle drive', {
+        drive,
+        previouslySelected: selectionState.isCurrentDrive(availableDrives.device),
+        applicationSessionUuid: store.getState().toJS().applicationSessionUuid,
+        flashingWorkflowUuid: store.getState().toJS().flashingWorkflowUuid
+      })
+
+      selectionState.toggleDrive(drive.device)
+    }
+  }
+
+  /**
+   * @summary Prompt the user to install missing usbboot drivers
+   * @function
+   * @public
+   *
+   * @param {Object} drive - drive
+   * @returns {void}
+   *
+   * @example
+   * installMissingDrivers({
+   *   linkTitle: 'Go to example.com',
+   *   linkMessage: 'Examples are great, right?',
+   *   linkCTA: 'Call To Action',
+   *   link: 'https://example.com'
+   * });
+   */
+  const installMissingDrivers = (drive) => {
+    if (drive.link) {
+      analytics.logEvent('Open driver link modal', {
+        url: drive.link,
+        applicationSessionUuid: store.getState().toJS().applicationSessionUuid,
+        flashingWorkflowUuid: store.getState().toJS().flashingWorkflowUuid
+      })
+
+      setConfirmModal({
+        open: true,
+        confirmationLabel: 'Yes, continue',
+        rejectionLabel: 'Cancel',
+        title: drive.linkTitle,
+        message: drive.linkMessage || `Etcher will open ${drive.link} in your browser`,
+        cancel: () => setConfirmModal({ open: false }),
+        done: async (shouldContinue) => {
+          try {
+            if (shouldContinue) {
+              openExternal(drive.link)
+            } else {
+              setConfirmModal({ open: false })
+            }
+          } catch (error) {
+            analytics.logException(error)
+          }
+        }
+      })
+    }
+  }
+
+  /**
+   * @summary Select a drive and close the modal
+   * @function
+   * @public
+   *
+   * @param {Object} drive - drive
+   * @param {Function} closeFn - close function
+   * @returns {void}
+   *
+   * @example
+   * selectDriveAndClose({
+   *   device: '/dev/disk2',
+   *   size: 999999999,
+   *   name: 'Cruzer USB drive'
+   * });
+   */
+  const selectDriveAndClose = async (drive, closeFn) => {
+    const canChangeDriveSelectionState = await shouldChangeDriveSelectionState(drive)
+
+    if (canChangeDriveSelectionState) {
+      selectionState.selectDrive(availableDrives.device)
```

Comment:
This probably should be `selectionState.selectDrive(drive.device)`.

Do we really want this behaviour ?
Double adds the drive to the selection even if it was already selected.

Reply:
> This probably should be

Correct, fixing